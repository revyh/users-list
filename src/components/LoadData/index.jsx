import React from 'react';
import CircularProgress from 'material-ui/CircularProgress';
import Snackbar from 'material-ui/Snackbar';

const progressStyles = {
  display: 'block',
  marginLeft: 'auto',
  marginRight: 'auto',
};

export default class LoadData extends React.Component {
  static defaultProps = {
    loadState: 'success',
  };

  componentDidMount() {
    if (this.props.loadState !== 'loading')
      this.handleLoad();
  }

  handleLoad = () => {
    this.props.onLoad && this.props.onLoad();
  };

  render() {
    const {loadState, children} = this.props;

    if (loadState === 'loading')
      return (
        <CircularProgress
          style={progressStyles}
          size={80}
          thickness={5}
        />
      );

    return (
      <div>
        {children}
        <Snackbar
          open={loadState === 'fail'}
          message="Sorry, something went wrong."
          action="Try again"
          onActionTouchTap={this.handleLoad}
        />
      </div>
    );
  }
}
