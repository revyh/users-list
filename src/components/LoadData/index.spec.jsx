import React from 'react';
import {shallow, mount} from 'enzyme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import LoadData from '.';

function mountWithContext(node) {
  return mount(
    <MuiThemeProvider muiTheme={getMuiTheme()}>{node}</MuiThemeProvider>,
  );
}

it('should render', () => {
  expect(shallow(<LoadData>some content</LoadData>)).toMatchSnapshot();
});

it('should render loading', () => {
  expect(shallow(<LoadData loadState="loading">some content</LoadData>))
    .toMatchSnapshot();
});

it('should render failed', () => {
  expect(shallow(<LoadData loadState="fail">some content</LoadData>))
    .toMatchSnapshot();
});

it('should trigger \'onLoad\' on mount', () => {
  const mockOnLoad = jest.fn();

  mountWithContext(<LoadData onLoad={mockOnLoad}>some content</LoadData>);

  expect(mockOnLoad).toHaveBeenCalled();
});

it('should not trigger \'onLoad\' on mount if data already loading', () => {
  const mockOnLoad = jest.fn();

  mountWithContext(
    <LoadData loadState="loading" onLoad={mockOnLoad}>some content</LoadData>,
  );

  expect(mockOnLoad).not.toHaveBeenCalled();
});

it('should trigger \'onLoad\' on fail', () => {
  const mockOnLoad = jest.fn();

  const loadData = shallow(
    <LoadData loadState="fail" onLoad={mockOnLoad}>some content</LoadData>,
  );

  loadData.children('Snackbar').simulate('actionTouchTap');

  expect(mockOnLoad).toHaveBeenCalled();
});
