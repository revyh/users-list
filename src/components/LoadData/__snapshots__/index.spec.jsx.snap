// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  some content
  <Snackbar
    action="Try again"
    message="Sorry, something went wrong."
    onActionTouchTap={[Function]}
    open={false}
  />
</div>
`;

exports[`should render failed 1`] = `
<div>
  some content
  <Snackbar
    action="Try again"
    message="Sorry, something went wrong."
    onActionTouchTap={[Function]}
    open={true}
  />
</div>
`;

exports[`should render loading 1`] = `
<CircularProgress
  max={100}
  min={0}
  mode="indeterminate"
  size={80}
  style={
    Object {
      "display": "block",
      "marginLeft": "auto",
      "marginRight": "auto",
    }
  }
  thickness={5}
  value={0}
/>
`;
